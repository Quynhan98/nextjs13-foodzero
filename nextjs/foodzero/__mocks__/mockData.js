export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=0a725961-6c86-4dcd-ab48-e5d7e2eb48fd',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  comments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=4e7dcf6c-2a66-4695-917d-af1d578d9a30',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
    reservations: [],
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=f2607e25-62ae-4ed9-9f53-bb76e6f830fb',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=9d70884c-e075-4f36-bf67-b196493badb8',
  },
]

export const OUR_MENU_MOCK = [
  {
    id: '1',
    name: 'Deep Sea Snow White Cod Fillet',
    price: 20,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit1.',
  },
  {
    id: '2',
    name: 'Steak With Rosemary Butter',
    price: 22,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
  },
  {
    id: '3',
    name: 'Cucumber Salad',
    price: 18,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
  },
  {
    id: '4',
    name: 'Natural Wine Pairing',
    price: 90,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit4.',
  },
]

export const BLOG_SECTION_MOCK = [
  {
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
    title: 'Fruit and vegetables and protection against diseases',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit1.',
    category: 'Fashion',
    srcAvatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjulie.webp?alt=media&token=c429d5f2-6f41-4233-970b-b2739746f7f1',
    name: 'Julie Christies',
    date: '2021-10-17T23:29:35.000Z',
    comments: 2,
    id: '1',
  },
  {
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fasparagus.webp?alt=media&token=d56ab468-1479-4181-844d-f5b231742244',
    title: "Asparagus Spring Salad with Rocket, Goat's Cheese",
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
    category: 'Fashion',
    srcAvatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fdianne.webp?alt=media&token=ce3d7213-72ce-4caa-b43c-f6f07bb3b5b7',
    name: 'Dianne Russell',
    date: '2021-10-17T23:29:35.000Z',
    comments: 2,
    id: '2',
  },
]

export const MENU_MOCK = {
  id: '1',
  ourMenu: OUR_MENU_MOCK,
  starters: OUR_MENU_MOCK,
  mains: OUR_MENU_MOCK,
  pastriesAndDrinks: OUR_MENU_MOCK,
}
