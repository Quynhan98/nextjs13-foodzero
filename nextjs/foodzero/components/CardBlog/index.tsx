import Image from 'next/image'
import { memo } from 'react'
import { Box, BoxProps, Heading, Text } from '@chakra-ui/react'

// Component
import BlogAvatar from '@components/BlogAvatar'

// Themes
import { rufina } from '@themes/index'

interface CardBlogProps extends BoxProps {
  imageUrl: string
  title: string
  description: string
  category: string
  srcAvatar: string
  name: string
  date: string
  comments: number
}

const CardBlog = ({
  imageUrl,
  title,
  description,
  category,
  srcAvatar,
  name,
  date,
  comments,
  ...rest
}: CardBlogProps) => {
  return (
    <Box position="relative" {...rest}>
      <Box
        position="absolute"
        top="12px"
        right="12px"
        background="bronzeYellow"
        textAlign="center"
        padding={{ base: '5px 20px', '2xl': '7px 53px' }}
      >
        <Text
          size="large"
          variant="primary"
          fontFamily={rufina}
          fontWeight="bold"
        >
          {category}
        </Text>
      </Box>
      <Image src={imageUrl} alt="blog wallpaper" width={792} height={792} />
      <Box
        padding={{ base: '30px 15px', '2xl': '68px 75px' }}
        textAlign="center"
        margin="0 auto"
      >
        <BlogAvatar
          src={srcAvatar}
          name={name}
          date={date}
          numberOfComments={comments}
        />
        <Heading
          as="h4"
          size="base"
          variant="secondary"
          pb="18px"
          pt={{ base: '20px', '2xl': '48px' }}
          textAlign="start"
        >
          {title}
        </Heading>
        <Box as="hr" color="black" borderTop="4px dotted black" />
        <Text size="default" pt="18px" textAlign="start">
          {description}
        </Text>
      </Box>
    </Box>
  )
}

export default memo(CardBlog)
