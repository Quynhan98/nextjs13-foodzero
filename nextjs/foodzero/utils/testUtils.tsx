import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
