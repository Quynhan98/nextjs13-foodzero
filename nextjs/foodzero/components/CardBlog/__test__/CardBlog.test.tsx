import { render, screen } from '@utils/testUtils'

// Mocks
import { CARD_BLOG } from '@mocks/mockData'

// Components
import CardBlog from '@components/CardBlog'

describe('CardBlog component', () => {
  const props = { ...CARD_BLOG }

  it('Renders CardBlog correctly', () => {
    const { container } = render(<CardBlog {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render data correctly', () => {
    render(<CardBlog {...props} />)

    const title = screen.getByRole('heading', { level: 4 })
    const category = screen.getByText(props.category)
    const name = screen.getByText(props.name)

    expect(title).toBeTruthy()
    expect(category).toBeTruthy()
    expect(name).toBeTruthy()
  })
})
