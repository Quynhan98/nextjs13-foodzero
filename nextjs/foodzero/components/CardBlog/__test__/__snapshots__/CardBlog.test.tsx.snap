// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBlog component Renders CardBlog correctly 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <div
      class="css-1frtpwp"
    >
      <p
        class="chakra-text css-1148kbg"
      >
        Fashion
      </p>
    </div>
    <img
      alt="blog wallpaper"
      data-nimg="1"
      decoding="async"
      height="792"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D0a725961-6c86-4dcd-ab48-e5d7e2eb48fd&w=1920&q=75"
      srcset="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D0a725961-6c86-4dcd-ab48-e5d7e2eb48fd&w=828&q=75 1x, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D0a725961-6c86-4dcd-ab48-e5d7e2eb48fd&w=1920&q=75 2x"
      style="color: transparent;"
      width="792"
    />
    <div
      class="css-1w5suxd"
    >
      <div
        class="css-70qvj9"
      >
        <span
          class="chakra-avatar css-em63j9"
        >
          <div
            aria-label="Julie Christie"
            class="chakra-avatar__initials css-184jamk"
            role="img"
          >
            JC
          </div>
        </span>
        <ul
          class="css-91sxya"
          role="list"
        >
          <li
            class="css-1xi7oxx"
          >
            Julie Christie
          </li>
          <li
            class="css-8ri12f"
          >
            December 06, 2022
          </li>
          <li
            class="css-8ri12f"
          >
            06:29 AM
          </li>
          <li
            class="css-1rh9xtx"
          >
            2
             comments
          </li>
        </ul>
      </div>
      <h4
        class="chakra-heading css-28jvh4"
      >
        Fruit and vegetables and protection against diseases
      </h4>
      <hr
        class="css-113h5p7"
      />
      <p
        class="chakra-text css-17gtsc2"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
